documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,157 @@
+# Repository Guidelines for Cadl Specs
+
+## Purpose
+
+We need to formulate a strategy for checking in actual Cadl service specs for Azure. Service teams need to know where
+and how to check these in and tooling needs to know how to find and consume them. While OpenAPI specs traditionally have been just spec
+documents, Cadl "specs" could include service-specific functions, decorators and so forth that elevate them to the level of a
+library, and to maximize the benefits of Cadl, this approach needs to be supported as a first-class citizen.
+
+## Repository
+
+Cadl can co-exist with Swagger within the existing `azure-rest-api-specs` repository. This approach will make it easier to generate Swagger artifacts without needing to sync between repos at the downside of having to live with any baggage associated with the repo as it ages. There are currently over 1,000 issues and 500 open PRs in this repo.
+
+## Structure Overview
+
+This proposal strives to align with [Azure SDK Repo Structure Guidelines](https://azure.github.io/azure-sdk/policies_repostructure.html).
+
+We eliminate the distinction between data-plane and resource-manager, treating all packages as siblings under the service family folder. This is what the SDKs do today. The legacy `data-plane` and `resource-manager` folders where Swagger specs live will remain.
+
+A given Cadl folder could represent any of the following scenarios:
+
+- an SDK and service spec (simplest, common for small services)
+- a service spec only
+- an SDK spec that aggregates many small service specs (one SDK to many microservices)
+- an SDK spec that extracts a portion of a large service spec (many SDKs to a single large service)
+- a shared library that represents neither a service spec nor an SDK
+
+Examples below:
+
+```
+-> specification
+   -> confidentialledger
+      -> ConfidentialLedger               (data-plane SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> compute
+      -> Compute.Management               (mgmt SDK + service)
+      -> Compute.Management.Shared        (shared)
+      -> Compute.CloudService.Management  (service)
```

Comment:
Can we consider any folder name that includes 'Management' is mgmt cadl? Or 'Management' has to be in the end or be second word in folder name? Can we have this clearly documented?

Reply:
The proposal says that management libraries end with "Management" unless it is a shared management library, in which case it ends in "Management.Shared".